src/watchdog/src/watchdog.py

```diff
@@ -241,11 +241,14 @@ def _map_fn(item):
     map(_map_fn, podsJsonObject["items"])
 
 
-def collect_healthz(gauge, histogram, service_name, address, port, url):
+def collect_healthz(gauge, histogram, service_name, address, port, url, ca_path, headers):
     with histogram.time():
         error = "ok"
         try:
-            error = requests.get("http://{}:{}{}".format(address, port, url)).text
+            if (ca_path is not None) and (headers is not None):
+                error = requests.get("https://{}:{}{}".format(address, port, url), headers = headers, verify = ca_path).text
```

Comment:
Maybe `headers` and `verify` have default value None? If this is the case, I thinke we may not need an if else branch, `error = requests.get("https://{}:{}{}".format(address, port, url), headers = headers, verify = ca_path).text` should be ok for both cases.